nemo/collections/nlp/modules/common/megatron/megatron_bert.py

```diff
@@ -80,8 +80,9 @@ def _update_model_parallel_arg(parser):
         # Initialize part of Megatron global state that is needed for its constructor.
         # We set 'lazy_mpu_init' flag on to make Megatron do only the initialization that does not depend
         # on ddp be initialized yet (and we don't want Megatron to initialize DDP itself either)
-        # and to return a hook for us to call after PTL has torch.distributed initialized
-        # (that would be on our first forward() call).
+        # and to return a hook for us to call after PTL has torch.distributed initialized.
+        # We call this hook during .forward
+        # TODO: can we call this hook using the PTL hook .setup()
```

Comment:
Can this be tried? If it works, it simplifies this part,

Reply:
I tried it and it wasn't working with single GPU "model parallel". It's something we can investigate more.

Reply:
Sounds good. Let's leave it as is for now.